phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/containers/ContainerVerbsTab.java

```diff
@@ -0,0 +1,94 @@
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.Tab;
+import javafx.scene.layout.TilePane;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+import org.phoenicis.containers.dto.ContainerDTO;
+import org.phoenicis.engines.VerbsManager;
+import org.phoenicis.javafx.views.common.ErrorMessage;
+import org.phoenicis.javafx.views.common.TextWithStyle;
+import org.phoenicis.repository.dto.ApplicationDTO;
+import org.phoenicis.repository.dto.ScriptDTO;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.phoenicis.configuration.localisation.Localisation.tr;
+
+public class ContainerVerbsTab extends Tab {
+    private static final String CONFIGURATION_PANE_CSS_CLASS = "containerConfigurationPane";
+    private static final String TITLE_CSS_CLASS = "title";
+
+    private final ContainerDTO container;
+    private VerbsManager verbsManager;
+
+    private final List<Node> lockableElements = new ArrayList<>();
+
+    public ContainerVerbsTab(ContainerDTO container, VerbsManager verbsManager,
+            ApplicationDTO verbs) {
+        super(tr("Verbs"));
+
+        this.container = container;
+        this.verbsManager = verbsManager;
+
+        this.setClosable(false);
+
+        this.populate(verbs);
+    }
+
+    private void populate(ApplicationDTO verbs) {
+        final VBox verbsPane = new VBox();
+        final Text title = new TextWithStyle(tr("Verbs"), TITLE_CSS_CLASS);
+
+        verbsPane.getStyleClass().add(CONFIGURATION_PANE_CSS_CLASS);
+        verbsPane.getChildren().add(title);
+
+        final TilePane verbsContentPane = new TilePane();
+        verbsContentPane.setPrefColumns(3);
+        verbsContentPane.getStyleClass().add("grid");
+
+        for (ScriptDTO verb : verbs.getScripts()) {
+            Button verbButton = new Button(verb.getScriptName());
+            verbButton.getStyleClass().addAll("toolButton");
+            verbButton.setStyle("-fx-background-image: url('" + verb.getIcon() + "');");
+            verbButton.setOnMouseClicked(event -> {
+                this.lockAll();
+                // TODO: find a better way to get the engine ID
+                this.verbsManager.installVerb(container.getEngine().toLowerCase(), container.getName(), verb.getId(),
+                        this::unlockAll,
+                        e -> Platform.runLater(() -> new ErrorMessage("Error", e).show()));
+            });
+            this.lockableElements.add(verbButton);
+            verbsContentPane.getChildren().add(verbButton);
+        }
+
+        verbsPane.getChildren().add(verbsContentPane);
+
+        final ScrollPane scrollPane = new ScrollPane();
+        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER); // the TilePane adjusts the number of columns
+                                                                    // already
+        verbsContentPane.prefWidthProperty().bind(scrollPane.widthProperty());
```

Comment:
Does it work with `verbsContentPane.setPrefWidth(0)` or `verbsContentPane.setPrefWidth(-1)` too?
As far as I know this means, that the container should use the width of its parent.

Reply:
I'm not sure and I don't really want to change this now. If you want to change it, we should handle this in a separate issue and also fix the same code for engine tools at the same time.